homeassistant/components/history_stats/sensor.py

```diff
@@ -124,35 +124,40 @@ def __init__(
         self._name = name
         self._unit_of_measurement = UNITS[sensor_type]
 
-        self._period = (datetime.datetime.now(), datetime.datetime.now())
+        self._period = (datetime.datetime.min, datetime.datetime.min)
         self.value = None
         self.count = None
+        self._history_current_period: list[State] = []
+        self._previous_run_before_start = False
+
+    @callback
+    def _async_start_refresh(self, *_) -> None:
+        """Register state tracking."""
+        self.async_schedule_update_ha_state(True)
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, [self._entity_id], self._async_update_from_event
+            )
+        )
 
     async def async_added_to_hass(self):
         """Create listeners when the entity is added."""
-
-        @callback
-        def start_refresh(*args):
-            """Register state tracking."""
-
-            @callback
-            def force_refresh(*args):
-                """Force the component to refresh."""
-                self.async_schedule_update_ha_state(True)
-
-            force_refresh()
-            self.async_on_remove(
-                async_track_state_change_event(
-                    self.hass, [self._entity_id], force_refresh
-                )
-            )
-
         if self.hass.state == CoreState.running:
```

Comment:
We can use the `async_at_start` helper here: https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/start.py#L11

Reply:
78ba096db1e1cf3a6a39f92c17e6fd15bf78492f